pumpkin-lib/src/engine/constraint_satisfaction_solver.rs

```diff
@@ -542,7 +544,14 @@ impl ConstraintSatisfactionSolver {
     }
 
     pub fn log_statistics(&self) {
-        self.counters.log_statistics()
+        self.counters.log(StatisticLogger::default());
+        for (index, propagator) in self.cp_propagators.iter_propagators().enumerate() {
+            propagator.log_statistics(StatisticLogger::new(format!(
```

Comment:
Here it would probably be tidier to get the prefix components (the strings separated by `_`) as some kind of slice of strings, and then do the concatenation within the constructor. That way the manner in which prefixes are concatenated is entirely encapsulated within the statistic logger.